Mathlib/Combinatorics/SimpleGraph/Connectivity.lean

```diff
@@ -2281,6 +2294,20 @@ def isoEquivSupp (φ : G ≃g G') (C : G.ConnectedComponent) :
   right_inv v := Subtype.ext_val (φ.toEquiv.right_inv ↑v)
 #align simple_graph.connected_component.iso_equiv_supp SimpleGraph.ConnectedComponent.isoEquivSupp
 
+lemma mem_coe_supp_of_adj {H : Subgraph G} {c : ConnectedComponent H.coe}
+    (hv : v ∈ (c.supp : Set V)) (hw : w ∈ H.verts)
+    (hadj : H.Adj v w) : w ∈ (c.supp : Set V) := by
+  rw [Set.mem_image]
```

Comment:
`SetLike` already provides the `Mem` instance for your convenience
```suggestion
lemma mem_supp_of_adj {H : Subgraph G} {c : ConnectedComponent H.coe}
    (hv : v ∈ c.supp) (hw : w ∈ H.verts)
    (hadj : H.Adj v w) : w ∈ c.supp := by
  rw [← SetLike.mem_coe, Set.mem_image]
```

Reply:
I think the thing is that `c.supp : Set ↑H.verts`, with your changes I get `failed to synthesize instance Membership ?m.219094 (Set ↑H.verts)`, is there a cleaner way to resolve this?

Reply:
Oh I see. `(c.supp : Set V)` inserts the coercion `Set ↑H.verts → Set V` (which is internal and not for use in mathlib). You should write that coercion explicitly instead by doing `v ∈ (↑) '' c.supp`

Reply:
Done:) Are these rules just for consistency or is there a technical reason to favour `SetLike.coe`?

Reply:
The technical reason is that otherwise people who try using your lemmas will have to deal with a function that appears nowhere else in mathlib, rather than the composition of two functions that each appear many times